lib/rubocop/cop/rspec/implicit_subject.rb

```diff
@@ -66,32 +94,70 @@ def on_send(node)
         private
 
         def autocorrect(corrector, node)
-          replacement = 'expect(subject)'
           case node.method_name
+          when :expect
+            corrector.replace(node, 'is_expected')
+          when :is_expected
+            corrector.replace(node.location.selector, 'expect(subject)')
           when :should
-            replacement += '.to'
+            corrector.replace(node.location.selector, 'expect(subject).to')
           when :should_not
-            replacement += '.not_to'
+            corrector.replace(node.location.selector, 'expect(subject).not_to')
           end
-
-          corrector.replace(node.loc.selector, replacement)
         end
 
-        def valid_usage?(node)
-          example = node.ancestors.find { |parent| example?(parent) }
-          return false if example.nil?
-
-          example.method?(:its) || allowed_by_style?(example)
+        def message(_node)
+          if style == :require_implicit
```

Comment:
Just to make the implementation parallel with `#invalid?`:
```suggestion
          case style
          when :require_implicit
```
…if RuboCop allows it.

Reply:
Oh, that's right. I'll change it like that 👍